src/Enum.php

```diff
@@ -19,7 +19,10 @@ public function __construct($name, $attribute = null, callable $resolveCallback
         parent::__construct($name, $attribute, $resolveCallback);
         $this->resolveUsing(
             function ($value) {
-                return $value instanceof UnitEnum ? $value->value : $value;
+                if($value instanceof BackedEnum){
+                    return $value->value;
+                }
+                return $value instanceof UnitEnum ? $value->name : $value;
```

Comment:
Hi, can you elaborate why in case of an `UnitEnum` the property `value` is no longer returned but `name`?

Do you have a code sample "`before`" "`after`" to highlight the necessity of this change? 

Did you make sure it doesn't break the current behavior?

Thanks.

Reply:
Hi, a simple UnitEnum does not value, its value is name. The current version does not support them. This change makes it possible to use them. Also, the JsonSerializable interface needs to be implemented for use in laravel.
Example:
`<?php
// simple UnitEnum, instance of UnitEnum
enum Status implements \JsonSerializable
{
    use LaravelEnumHelper;

    case ENABLED;
    case DISABLED;

    // need for laravel responses
    public function jsonSerialize(): string
    {
        // this condition is not needed here, the code is for common use (in a trait)
        if($this instanceof BackedEnum){
            return $this->value;
        }
        return $this->name;
    }
}

echo Status::ENABLED->value;

//result: PHP Warning:  Undefined property: Status::$value

// in model:
    protected $casts = [
        'status' => Status::class,
    ];`

My change doesn't change anything for BackedEnum.

`// instance of StringBackedEnum > BackedEnum > UnitEnum
enum StatusBackedString: string
{
    use LaravelEnumHelper;

    case ENABLED = 'enabled';
    case DISABLED = 'disabled';
}

// instance of IntBackedEnum > BackedEnum > UnitEnum
enum StatusBackedInt: int
{
    use LaravelEnumHelper;

    case ENABLED = 1;
    case DISABLED = 2;
}
`